### PR TITLE
DataFetcherConstructor args constructor as the annotation + tests

### PR DESCRIPTION
**Pull request by danieltaub96**
@GraphQLDataFetcher have args (String[]) fields, and the class thats implements DataFetcher should expect to get those args with String[] constructor...
of course that it's have Backwards compatibility
wrote some tests for the DataFetcherConstructor class

### CODE REVIEW

File: src/main/java/graphql/annotations/processor/util/DataFetcherConstructor.java [hunk 1]
```diff
@@ -38,14 +38,31 @@ public DataFetcher constructDataFetcher(String fieldName, GraphQLDataFetcher ann
...
+                    // filter only constructor that have the same args as the annotation args
+                    // or that get String[], String... args
+                    .filter(x -> (x.getParameterCount() == args.length) ||
+                            (x.getParameterTypes().length == 1 && x.getParameterTypes()[0] == String[].class))
+                    .map(x -> {
+                        try {
+                            Constructor<? extends DataFetcher> constructor;
+                            if (x.getParameterTypes().length == 1 && x.getParameterTypes()[0] == String[].class) {
+                                constructor = annotatedDataFetcher.value().getDeclaredConstructor(String[].class);
+                                return constructNewInstance(constructor, new Object[]{args});
```
Comment (guy120494) [inline src/main/java/graphql/annotations/processor/util/DataFetcherConstructor.java:51]: new Object[]{args} -> (Object[]) args

Comment (danieltaub96) [reply src/main/java/graphql/annotations/processor/util/DataFetcherConstructor.java:51]: it's in purpose that way 
new Object[]{args} create a single variable of object[],
and (Object[]) is a casting that casts to a object array, and the String[] ctor expecting only one argument...

**Comment by yarinvak** [comment]
Please udpate the readme with all the possible options of how the constructor of datafetche can look like

**Comment by danieltaub96** [comment]
done

**Pull request merged**